lua/ulx_yeet/sh_yeet.lua

```diff
@@ -0,0 +1,173 @@
+            if boneVec and boneAng then
+                bonePhys:SetPos( boneVec )
+                bonePhys:SetAngles( boneAng )
+            end
+            bonePhys:SetVelocity( velocity )
+        end
+    end
+
+    ply:Spectate( OBS_MODE_CHASE )
+    ply:SpectateEntity( ragdoll )
+    ply:StripWeapons()
+
+    return ragdoll
+end
+
+
+local function playerPickup( ply, ent )
+    local access, tag = ULib.ucl.query( ply, "ulx physgunplayer" )
+    if ent:IsPlayer() and ULib.isSandbox() and access and not ent.NoNoclip and not ent.frozen and ply:GetInfoNum( "cl_pickupplayers", 1 ) == 1 then
+        local restrictions = {}
+        ULib.cmds.PlayerArg.processRestrictions( restrictions, ply, {}, tag and ULib.splitArgs( tag )[1] )
+        if restrictions.restrictedTargets == false or (restrictions.restrictedTargets and not table.HasValue( restrictions.restrictedTargets, ent )) then
+            return
+        end
+
+        if CLIENT then return true end
+
+        ent:SetMoveType( MOVETYPE_NONE )
+        local newPos = ent:GetPos()
+        local oldPos = ent:GetPos()
+        local steamId = ent:SteamID64()
+
+        hook.Add( "Tick", "CFC_Yeet_TickHolding_" .. steamId, function()
+            if not IsValid( ent ) then
+                hook.Remove( "Tick", "CFC_Yeet_Tick_Holding" .. steamId )
+                return
+            end
+
+            newPos = ent:GetPos()
+            ent.cfcYeetSpeed = newPos - oldPos
+            oldPos = ent:GetPos()
+        end)
+
+        return true
+    end
+end
+hook.Add( "PhysgunPickup", "ulxPlayerPickup", playerPickup, HOOK_HIGH )
+
+if CLIENT then return end
+
+local function playerDrop( ply, ent )
+    if not ent:IsPlayer() then return end
+
+    hook.Remove( "Tick", "CFC_Yeet_Tick_Holding" .. ent:SteamID64() )
+
+    ent:SetMoveType( MOVETYPE_WALK )
+    ent:SetVelocity( ent.cfcYeetSpeed * 50 )
+
+    local access = ULib.ucl.query( ply, "physgunragdollplayer" )
+    if not access then return end
+
+    local speedLimit = GetConVar( "ulx_physgun_ragdoll_velocity" ):GetInt()
+
+    if ent.cfcYeetSpeed:Length() < speedLimit then return end
+
+    timer.Simple( 0, function()
+        local ragdoll = ragdollPlayer( ent, ent.cfcYeetSpeed * 50 )
+        ragdoll.player = ent
+        ragdoll.cooldown = CurTime() + 1
+
+        timer.Simple( 30, function()
+            unRagdollPlayer( ragdoll )
+        end)
+
+        local steamId = ent:SteamID64()
+        hook.Add( "Tick", "CFC_Yeet_Tick_Ragdoll" .. steamId, function()
+            if not IsValid( ragdoll ) then
+                hook.Remove( "Tick", "CFC_Yeet_Tick_Ragdoll" .. steamId )
+                ent:Spawn()
+                return
+            end
+
+            if ragdoll:GetVelocity():Length() > 10 or ragdoll.cooldown > CurTime() then return end
+            unRagdollPlayer( ragdoll )
+            hook.Remove( "Tick", "CFC_Yeet_Tick_Ragdoll" .. steamId )
+        end)
```

Comment:
Especially on Tick hooks, it's good practice to remove it ASAP - any code before it could error,  preventing the hook from being removed.

```suggestion
        local hookName = "CFC_Yeet_RagdollTick_" .. steamId

        hook.Add( "Tick", hookName, function()
            hook.Remove( "Tick", hookName )

            if not IsValid( ragdoll ) then
                ent:Spawn()
                return
            end

            if ragdoll:GetVelocity():Length() > 10 then return end
            if ragdoll.cooldown > CurTime() then return end

            unRagdollPlayer( ragdoll )
        end)
```